it/uk/gov/hmrc/agentmappingfrontend/controllers/MappingControllerISpec.scala

```diff
@@ -63,263 +71,78 @@ class MappingControllerISpec extends BaseControllerISpec with AuthStubs {
-      status(result) shouldBe 200
-    }
-
-    "redirect noArn go back to enter-account-number" in {
-      givenUserIsAuthenticated(eligibleAgent)
 
-      val result = callEndpointWith(request)
-      redirectLocation(result) shouldBe Some(routes.MappingController.showEnterAccountNo.url)
-    }
-  }
-
-  "submit add code" should {
-
-    val endpoint = "/agent-mapping/enter-utr"
-
-    behave like anEndpointReachableIfSignedInWithEligibleEnrolment(
-      POST,
-      endpoint,
-      expectCheckAgentRefCodeAudit = false)(callEndpointWith)
-
-    "redirect to complete if the user enters an ARN and UTR that match the known facts for SA" in {
-      givenUserIsAuthenticated(eligibleAgent)
-      mappingIsCreated(Utr("2000000000"), Arn("TARN0000001"))
-      val request = fakeRequest(POST, endpoint).withFormUrlEncodedBody("utr.value" -> "2000000000")
-        .withSession(("mappingArn", "TARN0000001"))
-      val result = callEndpointWith(request)
+        status(result) shouldBe 303
+        result.session.get("mappingArn").get shouldBe "TARN0000001"
+        redirectLocation(result).get should include(routes.MappingController.complete(id = "").url)
 
-      status(result) shouldBe 303
-      redirectLocation(result).get shouldBe routes.MappingController.complete().url
+        givenAuthorisedFor(serviceName)
+        val requestWithUsedIdShouldFail = fakeRequest(GET, s"/agent-mapping/start-submit?id=$persistedMappingArnResultId")
+        val resultCopiedAttempt = callEndpointWith(requestWithUsedIdShouldFail)
+        redirectLocation(resultCopiedAttempt) shouldBe Some(routes.MappingController.start().url)
+      }
     }
 
-    "redirect to complete if the user enters an ARN and UTR that match the known facts for VAT" in {
-      givenUserIsAuthenticated(vatEnrolledAgent)
-      mappingIsCreated(Utr("2000000000"), Arn("TARN0000001"))
-      val request =
-        fakeRequest(POST, endpoint).withFormUrlEncodedBody("utr.value" -> "2000000000").withSession(("mappingArn", "TARN0000001"))
+    s"303 to /account-linked persistedMappingArnResultId is INVALID" in {
+      givenAuthorisedFor("IR-SA-AGENT")
+      val request = fakeRequest(GET, s"/agent-mapping/start-submit?id=meaninglessBlaBlaID")
       val result = callEndpointWith(request)
-
       status(result) shouldBe 303
-      redirectLocation(result).get shouldBe routes.MappingController.complete().url
+      redirectLocation(result).get shouldBe routes.MappingController.start().url
     }
 
-    "redirect to the already-mapped page if the mapping already exists for SA" in new App {
-      givenUserIsAuthenticated(eligibleAgent)
-      mappingExists(Utr("2000000000"), Arn("TARN0000001"))
-
-      val request =
-        fakeRequest(POST, endpoint).withFormUrlEncodedBody("utr.value" -> "2000000000").withSession(("mappingArn", "TARN0000001"))
+    s"303 to /already-mapped when all available identifiers have been mapped" in {
+      val persistedMappingArnResultId = await(repo.create(arn))
```

Comment:
Are lines 115-122 needed for this test?

Reply:
a specific test, to get a realistic test for conflict.